src/prefect/engine.py

```diff
@@ -2158,14 +2158,16 @@ async def _run_task_hooks(task: Task, task_run: TaskRun, state: State) -> None:
 
 
 async def _run_flow_hooks(flow: Flow, flow_run: FlowRun, state: State) -> None:
-    """Run the on_failure and on_completion hooks for a flow, making sure to
+    """Run the on_failure, on_completion, and on_cancellation hooks for a flow, making sure to
     catch and log any errors that occur.
     """
     hooks = None
     if state.is_failed() and flow.on_failure:
         hooks = flow.on_failure
     elif state.is_completed() and flow.on_completion:
         hooks = flow.on_completion
+    elif state.is_cancelled() and flow.on_cancellation:
```

Comment:
What about `cancelling`?

Reply:
That's what I wrote initially 🤔 

I've changed it to `CANCELLING`!